connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -148,6 +148,7 @@ public KafkaStatusBackingStore(Time time, Converter converter) {
         this.statusTopic = statusTopic;
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
@rhauch @kkonstantine -- We deprecate `retries` via KIP-572. Not sure if we should rewrite this test right away so it does not use `retries` any longer?

Reply:
> Not sure if we should rewrite this test right away

First of all, this is not test code. It's actually part of the Connect runtime.

Second, I'd have to look into what we would do differently and how we would set `deliver.timeout.ms` to work with the existing logic that handles retries. Since it's not clear, I suggest we keep the current behavior of setting `retries=0` as long as that still function. We're using the `ProducerConfig.RETRIES` constant, so that can't be removed without changing this code.

Reply:
@rhauch @kkonstantine As discussed in person, I create a ticket that is marked as blocker for 2.7 release to address this issue: https://issues.apache.org/jira/browse/KAFKA-10297

Reply:
Btw: @hachikuji raise a concern about this issue, too: https://github.com/apache/kafka/pull/8864#pullrequestreview-443424531

> I just had one question about the proposal. Using retries=0 in the producer allows the user to have "at-most-once" delivery. This allows the application to implement its own retry logic for example. Do we still have a way to do this once this configuration is gone?

So maybe we need to do some follow up work in the `Producer` to make it work for Connect. But I would defer this to the follow up work.

My original though was, that setting `max.deliver.timeout.ms := request .timeout.ms` might prevent internal retries. But we need to verify this. It was also brought to my attention, that this might not work if the network disconnects -- only `retries=0` would prevent to re-open the connection but a low timeout would not prevent retries.

In KIP-572, we proposed for Kafka Streams itself, to treat `task.timeout.ms = 0` as "no retries" -- maybe we can do a similar thing for the producer?

There is also `max.block.ms` that we should consider. Unfortunately, I am not an expert on the `Producer`. But I would like to move forward with KIP-572 for now and are happy to help to resolve those questions.